src/block/block_compact.c

```diff
@@ -67,7 +67,7 @@ __wt_block_compact_skip(WT_SESSION_IMPL *session, WT_BLOCK *block, int *skipp)
 	 * worth doing.  Ignore small files, and files where we are unlikely
 	 * to recover 10% of the file.
 	 */
-	if (fh->size <= 10 * 1024)
+	if (fh->size <= 10 * WT_MEGABYTE)
```

Comment:
I agree it's arbitrary; I suspect this was intended to handle the case of lots and lots of small files where the user wanted to minimize total space.

What if we removed the `fh->size <= ###` test, and instead, did a minimum recoverable space check at the end of the function:
```
/*
 * Skip files where we can't recover at least 1MB.
 *
 * If at least 20% of the total file is available and in the first 80%
 * of the file, we'll try compaction on the last 20% of the file; else,
 * if at least 10% of the total file is available and in the first 90%
 * of the file, we'll try compaction on the last 10% of the file.
 *
 * We could push this further, but there's diminishing returns, a mostly
 * empty file can be processed quickly, so more aggressive compaction is
 * less useful.
 */
if (avail_ninety > WT_MEGABYTE && avail_ninety >= fh->size / 10) {
        ...
}
```
Does that make sense to you, instead of checking an arbitrary file size early on?

I agree we could make the size configurable through the API, but I think that can wait until we have a workload that needs it (and my bet is we never will).

Regardless, I don't think this test belongs in the btree layer, I think it's the block manager's problem.